app/src/main/java/org/mozilla/focus/browser/cursor/CursorViewModel.kt

```diff
@@ -75,32 +76,35 @@ class CursorViewModel(
     private fun update(deltaMillis: Long) {
         // Frames aren't guaranteed to occur at perfect intervals so we adjust the distance
         // travelled by the amount of time that has actually passed between frames (as
-        // opposed to the amount of time we expect to pass between frames): this should
-        // increase smoothness when the system can't keep up with our desired framerate.
+        // opposed to the amount of time we expect to pass between frames): this guarantees equal
+        // distance travelled when the system can't keep up with our desired framerate but the
+        // cursor will noticeably skip ahead as frames are dropped.
         //
-        // This adjustment could be expressed more naturally if this algorithm was expressed
-        // as a series of kinematic equations, i.e. vnew = vold + accel * deltaTime.
-        val deltaVel = (vel + 1) * FRICTION - vel
-        val timeAdjustedDeltaVel = deltaVel * (deltaMillis / UPDATE_DELAY_MILLIS.toFloat())
-        vel = Math.min(MAX_SPEED, vel + timeAdjustedDeltaVel)
```

Comment:
...but I'm also having trouble figuring out what parts are actually correct, especially if we're fixing the calculations in this review.

Reply:
We should pull this calculation into a different function like `getNewVelocity` and comment that the values and alg are experimentally obtained rather than adhering to any kinematic math.